File: frontend/src/app/models/liquid.ts

export class Liquid {
    name: string;
    type: LiquidType;
    co2: boolean;
    alcohol_percentage: number;
}

export enum LiquidType {
    Shotable = "SHOTABLE",
    Mixer = "MIXER",
    Syrup = "SYRUP"
}

File: backend/MixAPI/models/liquid.py
from django.db import models


class Liquid(models.Model):
    LiquidType = (
        ('SHOTABLE', 'SHOTABLE'),
        ('MIXER', 'MIXER'),
        ('SYRUP', 'SYRUP')
    )

    name = models.CharField(max_length=256, primary_key=True, unique=True)
    co2 = models.BooleanField(default=False)
    alcohol_percentage = models.FloatField()
    type = models.CharField(max_length=32, choices=LiquidType)

    def __str__(self):
        return self.name
